node/src/blockchain/blockchain_interface/blockchain_interface_web3/mod.rs

```diff
@@ -1046,7 +1051,37 @@ mod tests {
         assert_eq!(
             result,
             Ok(RetrievedBlockchainTransactions {
-                new_start_block: 1 + expected_fallback_start_block,
+                new_start_block: BlockNumber::Number((1 + expected_fallback_start_block).into()),
+                transactions: vec![]
+            })
+        );
+    }
+
+    #[test]
+    fn blockchain_interface_retrieve_transactions_start_and_end_blocks_can_be_latest() {
+        let port = find_free_port();
+        let _test_server = TestServer::start (port, vec![
+            br#"[{"jsonrpc":"2.0","id":1,"result":"error"},{"jsonrpc":"2.0","id":2,"result":[{"address":"0xcd6c588e005032dd882cd43bf53a32129be81302","blockHash":"0x1a24b9169cbaec3f6effa1f600b70c7ab9e8e86db44062b49132a4415d26732a","data":"0x0000000000000000000000000000000000000000000000000010000000000000","logIndex":"0x0","removed":false,"topics":["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef","0x0000000000000000000000003f69f9efd4f2592fd70be8c32ecd9dce71c472fc","0x000000000000000000000000adc1853c7859369639eb414b6342b36288fe6092"],"transactionHash":"0x955cec6ac4f832911ab894ce16aa22c3003f46deff3f7165b32700d2f5ff0681","transactionIndex":"0x0"}]}]"#.to_vec()
```

Comment:
I just don't understand what drove you to make the first transaction in the batch return an error. What does it imply in relation to the test? If it's actual the crucial peace of data in the response which set the right path in the code to go along it, maybe we should make it more obvious to the potential reader knowing about the context even less than for example me.  

Maybe you could do some gymnastics
like this: 

```Rust
let error_causing_to_pick_fallback_value = "error";
 let _test_server = TestServer::start (port, vec![
concat!(br#"[{"jsonrpc":"2.0","id":1,"result":"#,  error_causing_to_pick_fallback_value, br#"},{"jsonrpc":"2.0","id":2,"result":[{"address":"0xcd6c588e005032dd882cd43bf53a32129be81302" ..."#
```